src/components/ColorPalette/hooks/useCopyToClipboard.js

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+
+const useCopyToClipboard = () => {
+    const [copiedText, setCopiedText] = useState(null);
+    const [tooltip, setTooltip] = useState("Copy to clipboard");
```

Comment:
```suggestion
    const DEFAULT_COPY_TOOLTIP_TEXT = "Copy to clipboard";
    const [tooltip, setTooltip] = useState(DEFAULT_COPY_TOOLTIP_TEXT );
```
*OPTIONAL* sebenernya bisa di extract gini supaya orang yang baca codenya ngerti kalo itu default textnya. Walaupun sebenernya dimasukin ke useState sudah menjelaskan kalo itu default. You get to chose aja.